tendrl/ceph_integration/manager/cluster_monitor.py

```diff
@@ -366,6 +367,122 @@ def on_sync_object(self, data):
+                        raise rados.Error("Missing fields in cluster stat")
+                    index += 1
+                    if index >= len(lines):
+                        raise rados.Error("No cluster stats to parse")
+                    line = lines[index]
+                    cluster_fields = line.split()
+                    if len(cluster_fields) < 4:
+                        raise rados.Error("Missing fields in cluster stat")
+                    cluster_stat['total'] = self._to_bytes(
+                        cluster_fields[cluster_size_idx]
+                    )
+                    cluster_stat['used'] = self._to_bytes(
+                        cluster_fields[cluster_used_idx]
+                    )
+                    cluster_stat['available'] = self._to_bytes(
+                        cluster_fields[cluster_avail_idx]
+                    )
+                    cluster_stat['pcnt_used'] = cluster_fields[cluster_pcnt_used_idx]
+                if "POOLS" in line:
+                    pool_stat_available = True
+                    index += 1
+                    if index >= len(lines):
+                        raise rados.Error("No pool stats to parse")
+                    pool_fields = lines[index].split()
+                    pool_name_idx = self._idx_in_list(pool_fields, 'NAME')
+                    pool_id_idx = self._idx_in_list(pool_fields, 'ID')
+                    pool_used_idx = self._idx_in_list(pool_fields, 'USED')
+                    pool_pcnt_used_idx = self._idx_in_list(pool_fields, '%USED')
+                    pool_max_avail_idx = self._idx_in_list(pool_fields, 'MAX_AVAIL')
+                    if pool_name_idx == -1 or pool_id_idx == -1 or \
+                        pool_used_idx == -1 or pool_pcnt_used_idx == -1 or \
+                        pool_max_avail_idx == -1:
+                        raise rados.Error("Missing fields in pool stat")
+                    index += 1
+                if pool_stat_available:
+                    line = lines[index]
+                    pool_fields = line.split()
+                    if len(pool_fields) < 5:
+                        raise rados.Error("Missing fields in pool stat")
+                    dict = {}
+                    dict['name'] = pool_fields[pool_name_idx]
+                    dict['available'] = self._to_bytes(
+                        pool_fields[pool_max_avail_idx]
+                    )
+                    dict['used'] = self._to_bytes(
+                        pool_fields[pool_used_idx]
+                    )
+                    dict['pcnt_used'] = pool_fields[pool_pcnt_used_idx]
+                    pool_stat.append(dict)
+                index += 1
+            return {'cluster': cluster_stat, 'pools': pool_stat}
+
+    def _idx_in_list(self, list, str):
+        idx = -1
+        for item in list:
+            idx += 1
+            if item == str:
+                return idx
+        return idx
```

Comment:
As discussed, if the match is not found for the string in array, in last I would return -1.